include/ozo/transaction.h

```diff
@@ -136,8 +583,10 @@ constexpr auto get_transaction_mode(const impl::transaction<Ts...>& transaction)
  * @return ozo::none --- if no deferrability has been set
  */
 template<typename... Ts>
-constexpr auto get_transaction_deferrability(const impl::transaction<Ts...>& transaction) {
+constexpr auto get_transaction_deferrability(const transaction<Ts...>& transaction) {
     return get_option(transaction.options(), transaction_options::deferrability, none);
 }
 
 } // namespace ozo
+
+#include <ozo/impl/transaction.h>
```

Comment:
Is it possible to move transaction type to a different header to avoid this cyclic include dependecy?

Reply:
Well, actually there is no problem with circular dependency since the implementation header is included at the end of public header. And the public header is included at the begin of the implementation header. So in any case we have the same sequence of included headers and every header included only once.